File: message-bus/src/main/java/alma/obops/draws/messages/configuration/PersistedEnvelopeRepository.java
package alma.obops.draws.messages.configuration;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jdbc.repository.query.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Component;

import alma.obops.draws.messages.rabbitmq.PersistedEnvelope;

@Component
public interface PersistedEnvelopeRepository extends CrudRepository<PersistedEnvelope, String> {
	@Query( "select * "
			+ "from envelope "
			+ "where state = :state" )
	List<PersistedEnvelope> findByState( @Param("state") String state );

	@Query( "select * " + 
			"from envelope " + 
			"where envelope_id = :envelope_id" )
	Optional<PersistedEnvelope> findByEnvelopeId( @Param("envelope_id") String envelope_id );
}

File: message-bus/src/main/python/adapt/messagebus/configuration/PersistedEnvelopeRepository.py
class PersistedEnvelopeRepository(object):
    #@Query( "select * "
    #		+ "from envelope "
    #		+ "where state = :state" )
    def findByState(self, state):
        return None
    #@Query( "select * " + 
    #		"from envelope " + 
    #		"where envelope_id = :envelope_id" )
    def findByEnvelopeId(self, envelope_id):
        return None
    def deleteAll(self):
        pass
    def findAll(self):
        return []
    def save(self, envelope):
        pass
